infer/inference.cc

```diff
@@ -298,10 +313,14 @@ unique_ptr<cfg::CFG> Inference::run(core::Context ctx, unique_ptr<cfg::CFG> cfg)
                         typedSendCount++;
                     } else if (bind.bind.type.hasUntyped()) {
                         DEBUG_ONLY(histogramInc("untyped.sources", bind.bind.type.untypedBlame().rawId()););
-                        if (auto e = ctx.beginError(bind.loc, core::errors::Infer::UntypedValue)) {
-                            e.setHeader("This code is untyped");
-                            e.addErrorNote("Support for `{}` is minimal. Consider using `{}` instead.", "typed: strong",
-                                           "typed: strict");
+                        if (auto what = errorClassForUntyped(ctx.state, ctx.file)) {
+                            if (auto e = ctx.beginError(bind.loc, *what)) {
+                                e.setHeader("This code is untyped");
+                                if (*what == core::errors::Infer::UntypedValue) {
+                                    e.addErrorNote("Support for `{}` is minimal. Consider using `{}` instead.",
+                                                   "typed: strong", "typed: strict");
+                                }
+                            }
```

Comment:
(continued from above)

```suggestion
                        auto what = errorClassForUntyped(ctx.state, ctx.file);
                        if (auto e = ctx.beginError(bind.loc, what)) {
                            e.setHeader("This code is untyped");
                            if (what == core::errors::Infer::UntypedValue) {
                                e.addErrorNote("Support for `{}` is minimal. Consider using `{}` instead.",
                                               "typed: strong", "typed: strict");
                            }
```